yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,307 @@
+import base64
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_duration,
+    qualities,
+)
+
+
+class TxxxBaseIE(InfoExtractor):
```

Comment:
why have a base class?

Reply:
There are sites that seem to be from the same network due to the way they do base64 encoding is identical, but the web page does not follow the same pattern.

Example: https://porntop.com/video/101569/triple-threat-with-lia-lor-malena-morgan-and-dani-daniels/

Reply:
Will you be adding support for it too?

Reply:
I already have the some working code and I just added them to the branch.

porntop.com was rather awkward because the VideoObject object is declared inside a <script> tag, and with JavaScript code within the declaration.

I had to do some regex replacement for self._parse_json() to work.

pornzog.com just embed videos from other sites in their network.